client/request.go

```diff
@@ -159,3 +160,15 @@ func cancellableDo(ctx context.Context, client *http.Client, req *http.Request)
 		return resp.response, resp.err
 	}
 }
+
+func objectToJSONStream(obj interface{}) (io.Reader, error) {
+	if obj != nil {
+		b, err := json.Marshal(obj)
+		if err != nil {
+			return nil, err
+		}
+		return bytes.NewReader(b), nil
+	}
+
+	return nil, nil
```

Comment:
If the obj is nil, shall we return an error？
If we return `nil, nil`, the is.Reader is nil, then we need to judge if it is nil in the caller. WDYT?

Reply:
If the obj is `nil`, it means that we don't need to send the data to the server. HTTP Client will handle the nil case for us. It's ok I think. How do you think?

Reply:
OK, I am OK on this, just a remind to ensure this.